docs/pairing-architecture.md

```diff
@@ -0,0 +1,283 @@
+The Channel Server will now relay websocket messages from Mobile to Desktop and vice-versa.
+
+#### 3) Devices exchange metadata and show confirmation screens
+
+At this point the devices have established a baseline level of trust and can exchange metadata to allow the user to confirm the pairing on both sides. 
+
+* We use explicit confirmation screens to reduce the risk of phishing or over-the-shoulder attacks.
+
+The Mobile device sends what is essentially an OAuth request, delivered over the websocket channel rather than opened as a local webpage. 
+It includes information such as:
+```
+client_id - its registered OAuth client id
+state - OAuth state parameter
+scope - requested OAuth scopes
+code_challenge and code_challenge_method - OAuth PKCE parameters 
+keys_jwk - FxA Scoped Keys key-fetching public key
+```
+
+The Desktop device must carefully check the provided OAuth request and error out if anything seems amiss.
+
+The Desktop device sends an "account metadata" message that includes information about the account being connected to, such as:
+
+```
+the user's email address
+the user's profile picture and display name
+the device's display name.
+Additional server derived information potentially including:
+Counterpart's User Agent information
+Counterpart's GeoIP derived location
+```
+
+This information will help the Mobile device show a meaningful confirmation screen. 
+It's PII, but the channel has established a base level of trust between devices, so it should be safe to share this information directly.
+
+* Upon receiving the metadata message from the other device, both Desktop and Mobile show an explicit confirmation screen. 
+* There's no set confirmation order: the Supp and Auth will show the confirmations screens at the same time, and can be accepted in any order.
+
+For additional security, the Channel Server annotates each message with information about the originating device, such as geoip lookup data. This may help the user detect if they're connecting to a different device than expected.
+
+For additional security, each device can display the calculated confirmation code to the user. This may help the user detect if they're connecting to a different device than expected.
+
+#### 4) Desktop device authorizes OAuth request
+
+When the user accepts the confirmation screen on the Desktop device, it will use its local credentials to authorize the OAuth request from the Mobile device. This involves no user interaction, and will include the following steps:
+
+* Check that redirect_uri is the expected value; error out if not.
+* Check that scope contains exactly `"profile"` and `"https://identity.mozilla.com/apps/oldsync"`; error out if not.
+* Check that client_id is the expected value; error out if not.
+* Check that state exists and is well formed; error out if not.
+* Check that code_challenge and code_challenge_method exist and are well formed; error out if not.
```

Comment:
The desktop path doesn't appear to do any of these checks.  Should it, or are they being handled somewhere else?

Reply:
I think we can handle it to the OAuth relier, it already does it but only for Sign up it seems. I shall add it as a follow up